tests/test_repository.py

```diff
@@ -1770,3 +1770,87 @@ def test_get_touched_functions():
     touched_functions = repository.get_touched_functions(metrics["spaces"], [], [4],)
 
     assert touched_functions == {("outer_func", 1, 6), ("inner_func", 3, 5)}
+
+
+def test_get_commits():
+    # By default get_commits utilizes the following parameters:
+    # include_no_bug: bool = False
+    # include_backouts: bool = False
+    # include_ignored: bool = False
+
+    BACKOUT_COMMIT = "ec01c146f756b74d18e4892b4fd3aecba00da93e"
+    IGNORED_COMMIT = "7f27080ffee35521c42fe9d4025caabef7b6258c"
+    NOBUG_COMMIT = "75966ee1fe658b1767d7459256175c0662d14c25"
+    NOBUG_IGNORED_COMMIT = "75276e64701bfde7cf2dd1f851adfea6a92d5747"
+    NOBUG_IGNORED_BACKOUT_COMMIT = "46c1c161cbe189a59d8274e011085d76163db7f4"
+
+    retrieved_commits = list(repository.get_commits())
+    excluded_commits = [
+        IGNORED_COMMIT,
+        BACKOUT_COMMIT,
+        NOBUG_COMMIT,
+        NOBUG_IGNORED_COMMIT,
+        NOBUG_IGNORED_BACKOUT_COMMIT,
+    ]
+    # 10 mock commits, 1 ignored, 1 backouts, 1 no_bug, 1 no_bug and ignored, 1 no_bug, ignored, and backouts
+    assert len(retrieved_commits) == 5
+    assert not any(
+        excluded_commit in {c["node"] for c in retrieved_commits}
+        for excluded_commit in excluded_commits
+    )
+
+    retrieved_commits = list(repository.get_commits(include_backouts=True))
+    assert len(retrieved_commits) == 6
+    assert BACKOUT_COMMIT in (c["node"] for c in retrieved_commits)
+
+    retrieved_commits = list(repository.get_commits(include_ignored=True))
+    assert len(retrieved_commits) == 6
+    assert IGNORED_COMMIT in (c["node"] for c in retrieved_commits)
+
+    retrieved_commits = list(repository.get_commits(include_no_bug=True))
+    assert len(retrieved_commits) == 6
+    assert NOBUG_COMMIT in (c["node"] for c in retrieved_commits)
+
+    retrieved_commits = list(
+        repository.get_commits(include_ignored=True, include_backouts=True)
+    )
+    included_commits = [IGNORED_COMMIT, BACKOUT_COMMIT]
+    assert len(retrieved_commits) == 7
+    assert all(
+        {c["node"] for c in retrieved_commits} for included_commit in included_commits
+    )
```

Comment:
This assertion is not right, it is just asserting that `{c["node"] for c in retrieved_commits}` is not empty.
It should instead be:
```Python
assert all(included_commit in {c["node"] for c in retrieved_commits} for included_commit in included_commits)
```

Or better, if you make `included_commits` a set instead of a list (`included_commits = {IGNORED_COMMIT, BACKOUT_COMMIT}`), you can just do:
```Python
assert included_commits.issubset({c["node"] for c in retrieved_commits})
```

The same applies to the other assertions below.

Reply:
I've updated this and the following assertions to use `issubset` and thank you for the correction on the assert using the all syntax.